cypress/e2e/signIn.cy.js

```diff
@@ -12,23 +17,40 @@ describe('Sign In page', () => {
 
   before(() => {
     cy.task('db:clear');
+  });
+
+  beforeEach(() => {
     cy.task('generateUser').then((generateUser) => {
       user = generateUser;
+      signInPage.visit();
+      cy.register(user.email, user.username, user.password);
     });
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
-    signInPage.typeEmail(user.email);
-    signInPage.typePassword(user.password);
+    signInPage.fillLoginCredentials(user.email, user.password);
     signInPage.clickSignInBtn();
 
     homePage.assertHeaderContainUsername(user.username);
   });
 
-  it('should not provide an ability to log in with wrong credentials', () => {
+  it('should not provide an ability to log in with wrong email', () => {
+    const userEmail = '77' + user.email;
+    signInPage.fillLoginCredentials(userEmail, user.password);
+    signInPage.clickSignInBtn();
+
+    cy.contains(loginFailedMessage);
+    cy.contains(invalidUserCredentialsMessage);
+    cy.contains(confirmationMessage).click();
```

Comment:
You are using a stub/mock for the 'confirmationMessage'. Be cautious when using stubs and mocks as they can make your tests brittle and less reliable. Consider using real interactions whenever possible.

Reply:
Done